Add favorites and node searching to link modal

hickey opened the pull request:
This imports the jQuery select2 module to enable selecting from a list of favorite nodes and searching of the node database in the link command modal. This was inspired by supermon which allowed a list of favorite nodes to be created for quick linking without having to memorize node numbers. 

The favorites are enabled through a new configuration file, `favorites.ini`, and there is documentation in the sample file for users to create their own file. 

The link command modal has remained unchanged except for the node input box. This has now been changed to a selection input and then `select2` is initialized against the selection input. This is how the modal looks with this change.

<img width="1149" height="347" alt="image" src="https://github.com/user-attachments/assets/02e83ac7-352e-42f7-a727-d44f6fe60699" />

Clicking on the node selection box will display the list of favorites:

<img width="1144" height="337" alt="image" src="https://github.com/user-attachments/assets/f3878c2b-361e-46d4-a67c-42674b5a8300" />

Or if there is not a `favorites.ini` or no favorites defined for the node:

<img width="1142" height="341" alt="image" src="https://github.com/user-attachments/assets/53543694-acf5-4ec5-8d8a-ae02646b0c78" />

User is then able to start typing to search the node database and can enter a node number: 

<img width="1144" height="445" alt="image" src="https://github.com/user-attachments/assets/cc3381d8-6ee4-45f7-bcd4-c77e94b32e62" />

Or text to search the callsign, description and location fields of the node database:

<img width="1142" height="344" alt="image" src="https://github.com/user-attachments/assets/d50ad17a-23a5-4eff-bc56-1063972ea3e4" />

hickey committed:
Add favorites and node searching to link modal

Signed-off-by: Gerard Hickey <hickey@kinetic-compute.com>

hickey committed:
Add file watcher for favorites.ini

Signed-off-by: Gerard Hickey <hickey@kinetic-compute.com>

hickey commented:
Sorry for the multiple forced pushes. I had to setup the commit signing for the ssh key (much easier than the GPG signing I used to do but have not maintained for some time). 

I have just added a watcher task so that if the `favorites.ini` file is updated the list of favorites gets refreshed without having to restart the service.

hickey commented:
Closes #286

hickey committed:
fix: Recover from missing favorites.ini gracefully

Signed-off-by: Gerard Hickey <hickey@kinetic-compute.com>

hickey committed:
fix: Avoid favorites corruption when node IDs change

Signed-off-by: Gerard Hickey <hickey@kinetic-compute.com>

jxmx commented:
I'll have to take a look at this in detail but I'm consumed by work on the back-end upgrades at the moment. A few items from a quick look at the code changes:

* What happens in the node # box when the node isn't in a favorite and or in the database?
* It appears that this code is removing the ability to click on a row and have it auto-populate the node in command modal. That ability needs to be retained.
* Are you able to revert the commit(s) that pushed in a bunch of random whitespace changes? This patch is really hard to read with a bunch of unrelated linter-looking cleanups
* The various parts of the code and the packaging that handled licensing need to be updates since it appears you're adding MIT-licensed code into the system.

hickey commented:
**Item 1**: At one point the "unknown" node would be listed and would be used for the connect/disconnect, but  I think that was before I hooked up the node database. I just tried using an unknown node on my allmon3 instance and it would change the node number to the first favorite. I will update the search function to list the search term (i.e. the node number) as the first item returned from the search. This will allow the unkown node to stay in the select box for the connect/disconnect commands. 

**Item 2**: Ah... that explains the statement on the link command window: _Hint: clicking on a node line first will pre-populate node numbers as appropriate._ This statement is completely in the wrong place. I saw the statement as I was working on modifying the window but it made no sense as there was no nodes or anything to click on. Would it not be more apparent to people to have that statement on the main window associated with the list of nodes? By the time they see the statement they have already done something to bring up the link command window and the statement no longer applies. 

As for populating the selection box with the node number, that should not be a problem. I was just in the web developer extension trying to see how you are passing the node number, but the table is constantly being rewritten and I can not see the HTML. I will find it in the code and work out how it is being passed so that it can be inserted into the select box. 

**Item 3**: Actually it is the original code that has the white space at the end of line. If you notice all the changes are to remove the excess whitespace at the end of the lines. When I started working at GitLab I was instructed by the senior engineers looking at one of my merge requests to fix the settings in my editor to remove whitespace at the end of lines. For the first couple of minutes I really could not understand what the difference it would make. I quickly got educated and have always made sure that my editors are configured correctly to avoid adding spurious whitespace to lines in a file. 

GitLab has a handy toggle that will ignore whitespace changes if I remember correctly, but I don't seem to see this in the GitHub pull request. The only other way that I know of to ignore the whitespace changes is to use `git diff -w` and then only actual changes show up in the diff. 

I would advise you to make the same changes on your editors. Whitespace at the end of the line is just pure evil for Git or any other source code mangement platform for exactly what you are seeing. 

What can be done is to remove all the whitespace at the end of lines in all the files and commit that back on to the `develop` branch. I can than pull that branch and rebase my changes against that branch so that the whitespace changes fall out of the diff. You can use the following to remove all the end of line whitespace:

`find . -type f -exec sed -i 's/[[:space:]]*$//' {} \:`

The backslash semicolon is important and not a typo. It tells `find` where the end of the command is that it needs to execute. 

**Item 4**: Concerning the licensing, I would be happy to update the `README` and the Debian `copyright` file. Are there any other locations that you are concerned with?

hickey committed:
Update of copyright references

Signed-off-by: Gerard Hickey <hickey@kinetic-compute.com>

hickey committed:
Allow unknown nodes to be linked to

Signed-off-by: Gerard Hickey <hickey@kinetic-compute.com>

Pross committed:
Suppress parenthesis for transmit time when no time is yet accrued (#281)

During TX there is 1 second where the time is yet to be displayed so you end up with `Network Source ()` until the time is added.

You can also see this if you wait for a TX then refresh the page.

This commit just moves the parenthesis so they only show when there is a time to show.

Pross committed:
Add logged in status body class (#282)

jxmx commented:
Generally, it's in poor taste to merge a bunch of meaningless whitespace changes in with an actual feature/functionality request.

jxmx committed:
Merge branch 'develop' into favorites
